File: app/src/main/java/com/aayush/filmipedia/util/datasource/factory/CrewDataFactory.java
package com.aayush.filmipedia.util.datasource.factory;

import com.aayush.filmipedia.FilmipediaApplication;
import com.aayush.filmipedia.model.Crew;
import com.aayush.filmipedia.util.datasource.CrewDataSource;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;
import io.reactivex.disposables.CompositeDisposable;

public class CrewDataFactory extends DataSource.Factory<Long, Crew> {
    private MutableLiveData<CrewDataSource> mutableLiveData;
    private CrewDataSource crewDataSource;
    private FilmipediaApplication application;
    private CompositeDisposable compositeDisposable;

    private Long personId;

    public CrewDataFactory(FilmipediaApplication application,
                           CompositeDisposable compositeDisposable, Long personId) {
        this.application = application;
        this.compositeDisposable = compositeDisposable;
        this.personId = personId;
        mutableLiveData = new MutableLiveData<>();
    }

    @Override
    public DataSource<Long, Crew> create() {
        crewDataSource = new CrewDataSource(application, compositeDisposable, personId);
        mutableLiveData.postValue(crewDataSource);
        return crewDataSource;
    }

    public MutableLiveData<CrewDataSource> getMutableLiveData() {
        return mutableLiveData;
    }
}

File: app/src/main/java/com/aayush/filmipedia/util/datasource/factory/CrewDataFactory.kt
package com.aayush.filmipedia.util.datasource.factory

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import com.aayush.filmipedia.FilmipediaApplication
import com.aayush.filmipedia.model.Crew
import com.aayush.filmipedia.util.datasource.CrewDataSource
import io.reactivex.disposables.CompositeDisposable

class CrewDataFactory(
    private val application: FilmipediaApplication,
    private val compositeDisposable: CompositeDisposable,
    private val personId: Long?
): DataSource.Factory<Long, Crew>() {
    val mutableLiveData: MutableLiveData<CrewDataSource> = MutableLiveData()
    private lateinit var crewDataSource: CrewDataSource

    override fun create(): DataSource<Long, Crew> {
        crewDataSource = CrewDataSource(application, compositeDisposable, personId)
        mutableLiveData.postValue(crewDataSource)
        return crewDataSource
    }
}